src/tools/pkcs11-tool.c

```diff
@@ -4550,12 +4561,11 @@ static CK_RV write_object(CK_SESSION_HANDLE session)
 #if !defined(OPENSSL_NO_EC)
 		else if (pk_type == EVP_PKEY_EC)
 			type = CKK_EC;
-#ifdef EVP_PKEY_ED448
 		else if ((pk_type == EVP_PKEY_ED25519) || (pk_type == EVP_PKEY_ED448))
-#else
-		else if (pk_type == EVP_PKEY_ED25519)
-#endif
 			type = CKK_EC_EDWARDS;
+		else if ((pk_type == EVP_PKEY_ED25519) || (pk_type == EVP_PKEY_ED448))
```

Comment:
EVP_PKEY_X* ?

Reply:
This is comment is outdated, but there was a problem with `evp_pkey2ck_key_type` in this area that will be fixed.